drivers/SmartThings/zigbee-smoke-detector/src/aqara-gas/init.lua

```diff
@@ -0,0 +1,190 @@
+    minimum_interval = 1,
+    maximum_interval = 3600,
+    data_type = data_types.Uint16,
+    reportable_change = 1
+  }
+}
+
+local function gas_zone_status_handler(driver, device, value, zb_rx)
+  if value.value == 1 then
+    device:emit_event(capabilities.gasDetector.gas.detected())
+  elseif value.value == 0 then
+    device:emit_event(capabilities.gasDetector.gas.clear())
+  end
+end
+
+local function buzzer_status_handler(driver, device, value, zb_rx)
+  if value.value == 1 then
+    device:emit_event(capabilities.audioMute.mute.muted())
+  elseif value.value == 0 then
+    device:emit_event(capabilities.audioMute.mute.unmuted())
+  end
+end
+
+local function lifetime_status_handler(driver, device, value, zb_rx)
+  if value.value == 1 then
+    device:emit_event(lifeTimeReport.lifeTimeState.endOfLife())
+  elseif value.value == 0 then
+    device:emit_event(lifeTimeReport.lifeTimeState.normal())
+  end
+end
+
+local function selfcheck_status_handler(driver, device, value, zb_rx)
+  if value.value == 0 then
+    device:emit_event(selfCheck.selfCheckState.idle())
+  elseif value.value == 1 then
+    device:emit_event(selfCheck.selfCheckState.selfCheckCompleted())
+  end
+end
+
+local function sensitivity_adjustment_handler(driver, device, value, zb_rx)
+  if value.value == 0x01 then
+    device:emit_event(sensitivityAdjustment.sensitivityAdjustment.Low())
+  elseif value.value == 0x02 then
+    device:emit_event(sensitivityAdjustment.sensitivityAdjustment.High())
+  end
+end
+
+local function mute_handler(driver, device, cmd)
+  device:send(cluster_base.write_manufacturer_specific_attribute(device,
+    PRIVATE_CLUSTER_ID, PRIVATE_MUTE_ATTRIBUTE_ID, MFG_CODE, data_types.Uint8, 1))
+end
+
+local function unmute_handler(driver, device, cmd)
+  -- device:send(cluster_base.write_manufacturer_specific_attribute(device,
+  --   PRIVATE_CLUSTER_ID, PRIVATE_MUTE_ATTRIBUTE_ID, MFG_CODE, data_types.Uint8, 0))
+  device:emit_event(capabilities.audioMute.mute.muted())
+end
+
+local function sensitivity_adjustment_capability_handler(driver, device, command)
+  local sensitivity = command.args.sensitivity
+  local pre_sensitivity_value = 'High'
+
+  if pre_sensitivity_value ~= sensitivity then
+    if sensitivity == 'High' then
+    device:send(cluster_base.write_manufacturer_specific_attribute(device,
+      PRIVATE_CLUSTER_ID, PRIVATE_SENSITIVITY_ADJUSTMENT_ATTRIBUTE_ID, MFG_CODE, data_types.Uint8, 0x02))
+    elseif sensitivity == 'Low' then
+    device:send(cluster_base.write_manufacturer_specific_attribute(device,
+      PRIVATE_CLUSTER_ID, PRIVATE_SENSITIVITY_ADJUSTMENT_ATTRIBUTE_ID, MFG_CODE, data_types.Uint8, 0x01))
+    end
+    pre_sensitivity_value = sensitivity
+  else
+    if sensitivity == 'High' then
+      device:emit_event(sensitivityAdjustment.sensitivityAdjustment.High())
+    elseif sensitivity == 'Low' then
+      device:emit_event(sensitivityAdjustment.sensitivityAdjustment.Low())
```

Comment:
because of how this is structured, with the default value of `pre_sensitivity_value` always set to `High`, there's no way to hit this block

did you intend to get the current state of the sensitivityAdjustment attribute?

Reply:
EWEW

Reply:
Yes! The purpose of the mentioned code block is to avoid redundantly setting the same sensitivity. Prior to entering this branch, the code checks whether the current sensitivity value (sensitivity) is equal to the preset value (pre_sensitivity_value). If they are equal, it indicates that the sensitivity has not changed, and there is no need to resend the same configuration commands. Conversely, if the sensitivity value has changed, the code enters the previous conditional branch, executes the corresponding configuration command, and updates pre_sensitivity_value to the current sensitivity value for future comparisons. This ensures that the configuration commands are only executed when the sensitivity value truly changes, preventing unnecessary redundant operations.

Reply:
you should be able to get the most recent sensitivity value using `device:get_latest_state("main", sensitivityAdjustment.ID, sensitivityAdjustment.sensitivityAdjustment.NAME)`

Reply:
Thank you, it has been modified

Reply:
Can you write a test for this behavior?

You can expand your existing test that sets the sensitivity by adding a report from the device followed by another set to the same value. You would expect to see an immediate event.